benchmarks/Microsoft.AspNetCore.Routing.Performance/Template/TemplateSegmentBenchmark.cs

```diff
@@ -0,0 +1,38 @@
+﻿using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Routing.Patterns;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Routing.Template
+{
+    public class TemplateSegmentBenchmark
+    {
+        private RoutePatternPathSegment _segment;
+        private TemplateSegment _templateSegment;
+
+        [GlobalSetup]
+        public void GlobalSetup()
+        {
+            _segment = RoutePatternFactory.Segment(new List<RoutePatternPart>
+                {
+                    new RoutePatternSeparatorPart("a"),
+                    new RoutePatternSeparatorPart("c"),
+                    new RoutePatternSeparatorPart("e"),
+                    new RoutePatternSeparatorPart("f"),
+                    new RoutePatternSeparatorPart("g"),
+                });
+            _templateSegment = new TemplateSegment(_segment);
+        }
+
+        [Benchmark]
+        public void Ctor_RoutePatternPathSegment()
+        {
+            new TemplateSegment(_segment);
+        }
+
+        [Benchmark]
+        public void ToRoutePatternPathSegment()
+        {
+            _templateSegment.ToRoutePatternPathSegment();
+        }
```

Comment:
I'm really not sure that there's any performance sensitive code here

Reply:
Is this part of the code path from building a matcher?

Reply:
Pretty sure it's not performance sensitive, but any performance improvement that doesn't introduce complexity should be considered. No?

That said, I'll revisit this PR later this week to see if it's still worth it now that #855 will no longer eliminate the array copy.

Reply:
> Pretty sure it's not performance sensitive, but any performance improvement that doesn't introduce complexity should be considered

Sure, and believe me, I'm not complaining about getting something for free 😁 

There's a history here of people piling on this code because they understand it rather than the parts of the routing codebase that are perf-sensitive.